ShopAppTests/App/Cart/ViewModels/CartViewModelSpec.swift

```diff
@@ -0,0 +1,272 @@
+                }
+            }
+            
+            context("if cart product updated but error occured during loading cart products") {
+                it("should have error") {
+                    changeCartProductUseCaseMock.isNeedToReturnError = false
+                    cartProductListUseCaseMock.isNeedToReturnError = true
+                    cartProductListUseCaseMock.isNeedToReturnQuantity = false
+                    
+                    viewModel.update(cartProduct: cartProduct, quantity: 5)
+                    
+                    expect(viewModel.data.value.count) == 1
+                    expect(viewModel.data.value.first?.quantity) == 0
+                    expect(states.count) == 2
+                    expect(states.first) == ViewState.loading(showHud: true, isTranslucent: false)
+                    expect(states.last) == ViewState.error(error: nil)
+                }
+            }
+            
+            context("if cart product didn't update") {
+                it("should have error") {
+                    changeCartProductUseCaseMock.isNeedToReturnError = true
+                    cartProductListUseCaseMock.isNeedToReturnError = true
+                    cartProductListUseCaseMock.isNeedToReturnQuantity = false
+                    
+                    viewModel.update(cartProduct: cartProduct, quantity: 5)
+                    
+                    expect(viewModel.data.value.count) == 1
+                    expect(viewModel.data.value.first?.quantity) == 0
+                    expect(states.count) == 2
+                    expect(states.first) == ViewState.loading(showHud: true, isTranslucent: false)
+                    expect(states.last) == ViewState.error(error: nil)
+                }
+            }
+        }
+        
+        describe("when total price calculated") {
+            var cartProductFirst: CartProduct!
+            var cartProductSecond: CartProduct!
+            
+            beforeEach {
+                cartProductFirst = CartProduct()
+                cartProductFirst.quantity = 5
+                
+                let productVariantFirst = ProductVariant()
+                productVariantFirst.price = Decimal(floatLiteral: 10)
+                cartProductFirst.productVariant = productVariantFirst
+                
+                cartProductSecond = CartProduct()
+                cartProductSecond.quantity = 10
+                
+                let productVariantSecond = ProductVariant()
+                productVariantSecond.price = Decimal(floatLiteral: 15)
+                cartProductSecond.productVariant = productVariantSecond
+            }
+            
+            it("should calculate total price") {
+                viewModel.data.value = [cartProductFirst, cartProductSecond]
+                let totalPrice = viewModel.calculateTotalPrice()
+                
+                expect(totalPrice) == 5 * 10 + 10 * 15
```

Comment:
I think we can use concrete value for test and left math operations as comment, something like this:
 `expect(totalPrice) == 200 // 5 * 10 + 10 * 15`